Zend/zend_execute_API.c

```diff
@@ -491,6 +499,29 @@ ZEND_API const char *get_active_function_name(void) /* {{{ */
 }
 /* }}} */
 
+ZEND_API char *get_active_function_or_method_name(void) /* {{{ */
+{
+	if (!zend_is_executing()) {
+		return "";
+	}
+
+	return get_function_or_method_name(EG(current_execute_data)->func);
+}
+/* }}} */
+
+ZEND_API char *get_function_or_method_name(const zend_function *func) /* {{{ */
```

Comment:
I would go for zend_string * return value here. Use zend_string_copy for function names and zend_create_member_string for methods.

Reply:
I've changed the code to return a `zend_string *`, but I couldn't use `zend_string_copy()` without memory leaks, so I went with something slightly different. Unfortunately, even though I read http://www.phpinternalsbook.com/php7/internal_types/strings/zend_strings.html , I haven't quite understood when to use `0` or `1` in `zend_string_init()`. :(

Reply:
To be clear, using zend_string_copy requires using the zend_string* function_name. You can't recover it after it has been converted to char*.

Reply:
Ahh, at the first sight, it seemed that `get_function_name()` performs some logic that I shouldn't reimplement, but I've just understood that it's not required to use at all. Thank you for the suggestion!